uat/custom-components/src/main/java/com/aws/greengrass/artifacts/ShadowReactiveComponent.java

```diff
@@ -97,83 +90,45 @@ private boolean isNullOrEmpty(String s) {
+                            }
+                        } catch (IOException | ExecutionException | InterruptedException e) {
+                            LOGGER.error("Unable to send assertion for receive message", e);
                         }
-                    } catch (IOException | ExecutionException | InterruptedException e) {
-                        LOGGER.error("Unable to send assertion for receive message", e);
-                    }
 
-                    LATCH.countDown();
-                    LOGGER.info("Assertion posted");
-                }
-
-                @Override
-                public boolean onStreamError(Throwable error) {
-                    LOGGER.error("Received a stream error", error);
-                    return false;
-                }
-
-                @Override
-                public void onStreamClosed() {
-                    LOGGER.info("Subscribe to topic stream closed.");
-                }
-            })).getResponse();
-
-            try {
-                fut.get(50, TimeUnit.SECONDS);
-                LOGGER.info("Successfully subscribed to {}", topic);
-
-                if (!LATCH.await(180, TimeUnit.SECONDS)) {
-                    LOGGER.error("Timed out waiting for the message");
-                    updateComponentState(greengrassCoreIPCClient, ReportedLifecycleState.ERRORED);
-                }
-            } catch (TimeoutException e) {
-                LOGGER.error("Timeout occurred while subscribing to {}", topic, e);
-                updateComponentState(greengrassCoreIPCClient, ReportedLifecycleState.ERRORED);
-            } catch (ExecutionException e) {
-                if (e.getCause() instanceof UnauthorizedError) {
-                    UnauthorizedError unauthorizedError = (UnauthorizedError) e.getCause();
-                    LOGGER.error("Unauthorized error while subscribing to {}", topic, e);
-                    updateComponentState(greengrassCoreIPCClient, ReportedLifecycleState.ERRORED);
-                } else {
-                    LOGGER.error("Execution error while subscribing to {}", topic, e);
-                    updateComponentState(greengrassCoreIPCClient, ReportedLifecycleState.ERRORED);
+                        LATCH.countDown();
+                        messageReceived.set(true);
+                        LOGGER.info("Assertion posted");
+                    },
+                    Optional.of((error) -> {
+                        LOGGER.error("Received a stream error", error);
+                        return false;
+                    }),
+                    Optional.of(() -> {
+                        LOGGER.error("Subscribe to topic stream closed.");
+                    })).getResponse();
+            while (!messageReceived.get()) {
+                LOGGER.info("Waiting for message on topic for 5 seconds...");
+                if (LATCH.await(5, TimeUnit.SECONDS)) {
```

Comment:
else?

Reply:
We don't need `else` as I only want it to be blocked waiting for the message on callback and check for it every 5 seconds.